ui/components/src/TimeSeriesTooltip/TimeSeriesTooltip.tsx

```diff
@@ -11,78 +11,73 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Box, Portal, Typography, Stack, Switch } from '@mui/material';
+import { Box, Portal, Stack } from '@mui/material';
 import { ECharts as EChartsInstance } from 'echarts/core';
 import React, { useState } from 'react';
 import useResizeObserver from 'use-resize-observer';
 import { EChartsDataFormat, UnitOptions } from '../model';
 import { TooltipContent } from './TooltipContent';
+import { TooltipHeader } from './TooltipHeader';
 import { getNearbySeriesData } from './nearby-series';
 import {
   CursorCoordinates,
   FALLBACK_CHART_WIDTH,
+  TOOLTIP_BG_COLOR_FALLBACK,
   TOOLTIP_MAX_HEIGHT,
-  TOOLTIP_MIN_WIDTH,
   TOOLTIP_MAX_WIDTH,
+  TOOLTIP_MIN_WIDTH,
   useMousePosition,
 } from './tooltip-model';
 import { assembleTransform } from './utils';
 
-interface TimeSeriesTooltipProps {
+export interface TimeSeriesTooltipProps {
   chartRef: React.MutableRefObject<EChartsInstance | undefined>;
   chartData: EChartsDataFormat;
-  isTooltipPinned: boolean;
   wrapLabels?: boolean;
   unit?: UnitOptions;
   onUnpinClick?: () => void;
+  pinnedPos: CursorCoordinates | null;
 }
 
 export const TimeSeriesTooltip = React.memo(function TimeSeriesTooltip({
   chartRef,
   chartData,
   wrapLabels,
-  isTooltipPinned,
   unit,
   onUnpinClick,
+  pinnedPos,
 }: TimeSeriesTooltipProps) {
   const [showAllSeries, setShowAllSeries] = useState(false);
-  const [pinnedPos, setPinnedPos] = useState<CursorCoordinates | null>(null);
   const mousePos = useMousePosition();
   const { height, width, ref: tooltipRef } = useResizeObserver();
 
+  const isTooltipPinned = pinnedPos !== null;
+
   if (mousePos === null || mousePos.target === null) return null;
 
   // Ensure user is hovering over a chart before checking for nearby series.
   if (pinnedPos === null && (mousePos.target as HTMLElement).tagName !== 'CANVAS') return null;
 
   const chart = chartRef.current;
-  const chartWidth = chart?.getWidth() ?? FALLBACK_CHART_WIDTH; // Fallback width not likely to every be needed.
+  const chartWidth = chart?.getWidth() ?? FALLBACK_CHART_WIDTH; // Fallback width not likely to ever be needed.
   const cursorTransform = assembleTransform(mousePos, chartWidth, pinnedPos, height ?? 0, width ?? 0);
 
   // Get series nearby the cursor and pass into tooltip content children.
-  const focusedSeries = getNearbySeriesData({
+  const nearbySeries = getNearbySeriesData({
     mousePos,
     chartData,
     pinnedPos,
     chart,
     unit,
     showAllSeries,
   });
-  if (focusedSeries.length === 0) {
-    return null;
-  }
 
-  if (isTooltipPinned === true && pinnedPos === null) {
-    setPinnedPos(mousePos);
+  const seriesTimeMs = nearbySeries[0]?.date ? nearbySeries[0].date : null;
```

Comment:
Am I wrong to think we could shorten to this: `const seriesTimeMs = nearbySeries[0]?.date ?? null`